hazelcast/src/main/java/com/hazelcast/core/IMap.java

```diff
@@ -2588,4 +2588,25 @@ boolean tryLock(K key, long time, TimeUnit timeunit, long leaseTime, TimeUnit le
      */
     @Beta
     QueryCache<K, V> getQueryCache(String name, MapListener listener, Predicate<K, V> predicate, boolean includeValue);
+
+    /**
+     * Updates TTL (time to live) value of the entry specified by {@code key} with a new TTL value.
+     * New TTL value is valid from this operation is invoked, not from the original creation of the entry.
+     * <p>
+     * The entry will expire and get evicted after the TTL. If the TTL is 0,
+     * then the entry lives forever. If the TTL is negative, then the TTL
+     * from the map configuration will be used (default: forever).
+     *
+     * If there is no entry with key {@code key}, this call has no effect.
+     *
+     * <b>Warning:</b>
+     * <p>
+     * Time resolution for TTL is seconds. The given TTL value is rounded to the next closest second value.
+     *
+     * @param key      the key of the map entry
+     * @param ttl      maximum time for this entry to stay in the map (0 means infinite, negative means map config default)
+     * @param timeunit time unit for the TTL
+     * @since 3.11
+     */
+    void setTTL(K key, long ttl, TimeUnit timeunit);
```

Comment:
we can also add which exceptions can be thrown:
`@throws NullPointerException if the specified key or timeunit is null`